pathio.go

```diff
@@ -121,16 +108,31 @@ func parseS3Path(path string) (string, string, error) {
 	return bucketName, key, nil
 }
 
+// s3ConnectionInformation parses the s3 path and returns the s3 connection from the
+// correct region, as well as the bucket, and key
+func s3ConnectionInformation(path string) (s3Connection, error) {
+	bucket, key, err := parseS3Path(path)
+	if err != nil {
+		return s3Connection{}, err
+	}
+
+	// Look up region in S3
+	region, err := getRegionForBucket(newS3Handler("us-west-1"), bucket)
+	if err != nil {
+		return s3Connection{}, err
+	}
+
+	return s3Connection{newS3Handler(region), bucket, key}, nil
+}
+
 // getRegionForBucket looks up the region name for the given bucket
-func getRegionForBucket(name string) (string, error) {
+func getRegionForBucket(svc s3Handler, name string) (string, error) {
 	// Any region will work for the region lookup, but the request MUST use
 	// PathStyle
-	config := aws.NewConfig().WithRegion("us-west-1").WithS3ForcePathStyle(true)
-	client := s3.New(config)
 	params := s3.GetBucketLocationInput{
 		Bucket: aws.String(name),
 	}
-	resp, err := client.GetBucketLocation(&params)
+	resp, err := svc.GetBucketLocation(&params)
 	if err != nil {
 		return "", fmt.Errorf("Failed to get location for bucket '%s', %s", name, err)
 	}
```

Comment:
I know this is not something you added, but can you just change the `us-east-1` and `us-west-1` strings to a const defined at the head of this file? And we should just default to using one of them. I'd suggest `us-west-1` since that's our default region.